src/core/mac/data_poll_sender.cpp

```diff
@@ -504,8 +544,88 @@ void DataPollSender::HandlePollTimer(Timer &aTimer)
 
 uint32_t DataPollSender::GetDefaultPollPeriod(void) const
 {
-    return Time::SecToMsec(Get<Mle::MleRouter>().GetTimeout()) -
-           static_cast<uint32_t>(kRetxPollPeriod) * kMaxPollRetxAttempts;
+    uint32_t period = Time::SecToMsec(Get<Mle::MleRouter>().GetTimeout());
+
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+    if (Get<Mac::Mac>().IsCslEnabled())
+    {
+        UpdateIfLarger(period, Time::SecToMsec(Get<Mac::Mac>().GetCslTimeout()));
```

Comment:
Csl Timeout should always be smaller than or equal to Router Timeout. If it somehow Csl Timeout is larger than Router Timeout, we shoud use the smaller value?

Reply:
My mistake here. Please ignore.